h2o-algos/src/main/java/hex/tree/SharedTreeModelWithContributions.java

```diff
@@ -98,25 +171,61 @@ public void map(Chunk chks[], NewChunk[] nc) {
       Object workspace = _treeSHAP.makeWorkspace();
 
       for (int row = 0; row < chks[0]._len; row++) {
-        for (int i = 0; i < chks.length; i++) {
-          input[i] = chks[i].atd(row);
-        }
-        for (int i = 0; i < contribs.length; i++) {
-          contribs[i] = 0;
-        }
-
+        fillInput(chks, row, input, contribs);
         // calculate Shapley values
         _treeSHAP.calculateContributions(input, contribs, 0, -1, workspace);
-        
+        doModelSpecificComputation(contribs);
```

Comment:
this is never called because you are overriding method map(Chunk chks[], NewChunk[] nc)